packages/app-config-writer/src/smartlinks-config/service.ts

```diff
@@ -0,0 +1,88 @@
+import { Agent as HttpsAgent } from 'https';
+import { cyan } from 'chalk';
+import type { AxiosBasicCredentials } from 'axios';
+import axios from 'axios';
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import { create as createStorage } from 'mem-fs';
+import type { ToolsLogger } from '@sap-ux/logger';
+import { FileName } from '@sap-ux/project-access';
+import type { ServiceConfig, Service, SystemDetailsResponse } from '../types';
+import { UrlParameters } from '../types';
+import { t } from '../i18n';
+import { readUi5ConfigTargets, readUi5DeployConfigTargets } from './ui5-yaml';
+
+/**
+ * @description Combines the service parameters and returns the service url
+ * @param service service information with optional sap-client
+ * @returns service url to be called for smartlinks config
+ */
+function getSmartLinksServiceUrl(service: ServiceConfig) {
+    let appendUrl = Object.values(UrlParameters).join('&');
+    if (service.client) {
+        appendUrl = `${appendUrl}&sap-client=${service.client}`;
+    }
+    return `${service.url}${appendUrl}`;
+}
+
+/**
+ * @description Sends a request to service and returns the result
+ * @param service service to be used for request
+ * @param credentials credentials for service
+ * @param logger logger to report info to the user
+ * @returns response from service
+ */
+export async function sendRequest(
+    service: ServiceConfig,
+    credentials?: AxiosBasicCredentials,
+    logger?: ToolsLogger
+): Promise<SystemDetailsResponse> {
+    const serviceUrl = getSmartLinksServiceUrl(service);
+    const httpsAgent = new HttpsAgent({ rejectUnauthorized: service.ignoreCertError });
+    const axiosConfig = { auth: credentials, httpsAgent };
+    try {
+        logger?.info(`${cyan(t('info.connectTo'))} ${serviceUrl}`);
+        const response = await axios.get(serviceUrl, axiosConfig);
```

Comment:
Why don't we use the `@sap-ux/axios-extension` module here? It has been create for simplifying access to SAP services

Reply:
Thanks, yes will look into that

Reply:
Using `@sap-ux/axios-extension` now. Very basic approach taken for first iteration.